src/chunkserver/datastore/file_pool.cpp

```diff
@@ -375,6 +410,133 @@ void FilePool::CleanWorker() {
     }
 }
 
+bool FilePool::PrepareFormat() {
+    uint64_t bytesPerPage = poolOpt_.fileSize + poolOpt_.metaFileSize;
+    uint64_t needSpace = (poolOpt_.filePoolSize /
+                        bytesPerPage - currentState_.chunkNum) * bytesPerPage;
+    uint64_t vaildSpace = 0;
+    curve::fs::FileSystemInfo finfo;
+    if (poolOpt_.filePoolSize / bytesPerPage < currentState_.chunkNum) {
+        LOG(WARNING) << "The number of chunks that are pre-allocated"
+                        "is less than the number of initialized chunks.";
+        LOG(INFO) << "It is no need to format chunks.";
+        formatStat_.preAllocateNum = 0;
+        formatStat_.allocateChunkNum = 0;
+        return true;
+    }
+
+    int r = fsptr_->Statfs(currentdir_, &finfo);
+    if (r != 0) {
+        LOG(ERROR) << "get disk usage info failed!";
+        return false;
+    }
+    vaildSpace = finfo.available;
+    LOG(INFO) << "free space = " << finfo.available
+            << ", total space = " << finfo.total
+            << ", need space = " << needSpace;
+    if (vaildSpace < needSpace) {
+        LOG(ERROR) << "disk free space not enough.";
+        return false;
+    }
+
+    formatStat_.preAllocateNum = needSpace / bytesPerPage;
+    formatStat_.allocateChunkNum = 0;
```

Comment:
Is there a bug here？

Reply:
There's no problem initializing it to zero. If we need to create a filepool which chunk number is 10, and there are already 3 chunks in this chunkfile pool,  then the preAllocateNum is 7, allocateChunkNum is 0.